src/ILCompiler/src/Program.cs

```diff
@@ -305,7 +307,7 @@ private int Run(string[] args)
                     // TODO: Wasm fails to compile some of the exported methods due to missing opcodes
                     if (!_isWasmCodegen)
                     {
-                        compilationRoots.Add(new ExportedMethodsRootProvider(module));
+                        compilationRoots.Add(new ExportedMethodsRootProvider(module, _exportsFile));
```

Comment:
If there are multiple assemblies specified in `InputFilePaths` that have `NativeCallable` methods in them, they're going to stomp over each other's writes into the exports file.

I would prefer a scheme where `ExportedMethodsRootProvider` has a public property that returns an `IEnumerable<string>` of all the exported symbols.

Once we're done adding to the `compilationRoots` list here, we pass the list of compilation roots to a helper method that does a `foreach` over the list of root provider, finds the list of exported methods from each `ExportedMethodsRootProvider`, and writes it to a file.

Reply:
Ah thanks for this, totally overlooked the multiple input file paths scenario. On what type do you suggest the helper method should be?

Reply:
> On what type do you suggest the helper method should be?

I would probably just add a `DefFileWriter.cs` or something like that to the ILCompiler project (it probably doesn't even need to be in the ILCompiler.Compiler project). I also wouldn't mind much if it was in Program.cs, but that file is starting to be a dumping ground for lots of short methods and maybe we should stop doing that.